app/services/TransportMeansIdentificationTypesService.scala

```diff
@@ -17,33 +17,55 @@
 package services
 
 import cats.data.NonEmptyList
+import config.Constants
 import config.Constants.MeansOfTransportIdentification.UnknownIdentification
 import connectors.ReferenceDataConnector
 import models.Index
-import models.reference.TransportMode.BorderMode
+import models.reference.TransportMode.InlandMode
 import models.reference.transport.transportMeans.TransportMeansIdentification
+import models.requests.DataRequest
+import play.api.mvc.AnyContent
 import uk.gov.hmrc.http.HeaderCarrier
 
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-class TransportMeansIdentificationTypesService @Inject() (referenceDataConnector: ReferenceDataConnector)(implicit ec: ExecutionContext) {
+class TransportMeansIdentificationTypesService @Inject() (referenceDataConnector: ReferenceDataConnector, transportModeCodesService: TransportModeCodesService)(
+  implicit ec: ExecutionContext
+) {
 
-  def getMeansOfTransportIdentificationTypes(index: Index, borderModeOfTransport: Option[BorderMode])(implicit
-    hc: HeaderCarrier
-  ): Future[Seq[TransportMeansIdentification]] =
-    referenceDataConnector.getMeansOfTransportIdentificationTypes().map(filter(_, index, borderModeOfTransport)).map(sort)
+  def getMeansOfTransportIdentificationTypes(index: Index, inlandModeOfTransport: Option[InlandMode])(implicit
+    hc: HeaderCarrier,
+    request: DataRequest[AnyContent]
+  ): Future[Seq[TransportMeansIdentification]] = inlandModeOfTransport match {
+    case Some(InlandMode(_, _)) =>
+      referenceDataConnector.getMeansOfTransportIdentificationTypes().flatMap(filter(_, index, Future.successful(inlandModeOfTransport)).map(sort))
+    case None =>
+      referenceDataConnector
+        .getMeansOfTransportIdentificationTypes()
+        .flatMap(
+          filter(
+            _,
+            index,
+            transportModeCodesService.getInlandModes().map {
+              inlandModes =>
+                inlandModes.find(_.code == request.userAnswers.departureData.Consignment.inlandModeOfTransport.getOrElse(Constants.Unknown))
```

Comment:
Just checking if we wanted to return this Constants.Unkown or should we just flatMap over  `request.userAnswers.departureData.Consignment.inlandModeOfTransport`. Sorry I'm not that familiar with the business logic here.

Reply:
No, you are absolutely right, this should be using flatMap. I will update now, good spot 👍